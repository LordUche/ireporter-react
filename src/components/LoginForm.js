import React from 'react';
import { Card, Form } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { appRef } from '../utils/refs';
import { authenticateUser } from '../redux/actions/auth';

class LoginForm extends React.Component {
  state = { email: '', password: '' };

  handleInputChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = e => {
    e.preventDefault();
    const { logIn } = this.props;
    logIn('login', this.state);
  };

  render() {
    const { email, password } = this.state;
    const { loading, loggedIn } = this.props;

    if (loggedIn) return <Redirect to="/profile" />;
    return (
      <div ref={appRef} className="auth__form card-form">
        <Card>
          <Card.Content>
            <Card.Header as="h1">Log In</Card.Header>
            <Form onSubmit={this.handleSubmit}>
              <Form.Input
                icon="at"
                iconPosition="left"
                type="email"
                placeholder="Email"
                fluid
                onChange={this.handleInputChange}
                value={email}
                name="email"
                transparent
                required
              />
              <Form.Input
                icon="lock"
                iconPosition="left"
                type="password"
                placeholder="Password"
                fluid
                onChange={this.handleInputChange}
                value={password}
                name="password"
                transparent
                required
              />
              <Form.Button loading={loading} type="submit" color="red" fluid>
                Log In
              </Form.Button>
            </Form>
          </Card.Content>
        </Card>
      </div>
    );
  }
}

LoginForm.propTypes = {
  logIn: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  loggedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  loading: state.auth.loading,
  loggedIn: state.auth.loggedIn,
});

export default connect(
  mapStateToProps,
  { logIn: authenticateUser }
)(LoginForm);
