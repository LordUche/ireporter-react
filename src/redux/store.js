import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { logger } from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const middleware = [thunk];

process.env.NODE_ENV === `development` && middleware.push(logger);

const initialState = { auth: { loggedIn: false, loading: false } };
const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware))
);

export default store;
