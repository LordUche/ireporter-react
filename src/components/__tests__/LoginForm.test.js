import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import LoginForm from '../LoginForm';
import store from '../../redux/store';

describe('<LoginForm />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <LoginForm />
      </Provider>
    );
  });

  it('should match snapshot', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
