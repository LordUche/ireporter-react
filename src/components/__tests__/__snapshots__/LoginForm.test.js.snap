// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectFunction>
    <LoginForm
      loading={false}
      logIn={[Function]}
      loggedIn={false}
    >
      <div
        className="auth__form card-form"
      >
        <Card>
          <div
            className="ui card"
            onClick={[Function]}
          >
            <CardContent>
              <div
                className="content"
              >
                <CardHeader
                  as="h1"
                >
                  <h1
                    className="header"
                  >
                    Log In
                  </h1>
                </CardHeader>
                <Form
                  as="form"
                  onSubmit={[Function]}
                >
                  <form
                    className="ui form"
                    onSubmit={[Function]}
                  >
                    <FormInput
                      as={[Function]}
                      control={[Function]}
                      fluid={true}
                      icon="at"
                      iconPosition="left"
                      name="email"
                      onChange={[Function]}
                      placeholder="Email"
                      required={true}
                      transparent={true}
                      type="email"
                      value=""
                    >
                      <FormField
                        control={[Function]}
                        fluid={true}
                        icon="at"
                        iconPosition="left"
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        required={true}
                        transparent={true}
                        type="email"
                        value=""
                      >
                        <div
                          className="required field"
                        >
                          <Input
                            fluid={true}
                            icon="at"
                            iconPosition="left"
                            name="email"
                            onChange={[Function]}
                            placeholder="Email"
                            required={true}
                            transparent={true}
                            type="email"
                            value=""
                          >
                            <div
                              className="ui fluid transparent left icon input"
                            >
                              <input
                                name="email"
                                onChange={[Function]}
                                placeholder="Email"
                                required={true}
                                type="email"
                                value=""
                              />
                              <Icon
                                as="i"
                                name="at"
                              >
                                <i
                                  aria-hidden="true"
                                  className="at icon"
                                  onClick={[Function]}
                                />
                              </Icon>
                            </div>
                          </Input>
                        </div>
                      </FormField>
                    </FormInput>
                    <FormInput
                      as={[Function]}
                      control={[Function]}
                      fluid={true}
                      icon="lock"
                      iconPosition="left"
                      name="password"
                      onChange={[Function]}
                      placeholder="Password"
                      required={true}
                      transparent={true}
                      type="password"
                      value=""
                    >
                      <FormField
                        control={[Function]}
                        fluid={true}
                        icon="lock"
                        iconPosition="left"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        required={true}
                        transparent={true}
                        type="password"
                        value=""
                      >
                        <div
                          className="required field"
                        >
                          <Input
                            fluid={true}
                            icon="lock"
                            iconPosition="left"
                            name="password"
                            onChange={[Function]}
                            placeholder="Password"
                            required={true}
                            transparent={true}
                            type="password"
                            value=""
                          >
                            <div
                              className="ui fluid transparent left icon input"
                            >
                              <input
                                name="password"
                                onChange={[Function]}
                                placeholder="Password"
                                required={true}
                                type="password"
                                value=""
                              />
                              <Icon
                                as="i"
                                name="lock"
                              >
                                <i
                                  aria-hidden="true"
                                  className="lock icon"
                                  onClick={[Function]}
                                />
                              </Icon>
                            </div>
                          </Input>
                        </div>
                      </FormField>
                    </FormInput>
                    <FormButton
                      as={[Function]}
                      color="red"
                      control={[Function]}
                      fluid={true}
                      loading={false}
                      type="submit"
                    >
                      <FormField
                        color="red"
                        control={[Function]}
                        fluid={true}
                        loading={false}
                        type="submit"
                      >
                        <div
                          className="field"
                        >
                          <Button
                            as="button"
                            color="red"
                            fluid={true}
                            loading={false}
                            type="submit"
                          >
                            <Ref
                              innerRef={
                                Object {
                                  "current": <button
                                    class="ui red fluid button"
                                    type="submit"
                                  >
                                    Log In
                                  </button>,
                                }
                              }
                            >
                              <RefFindNode
                                innerRef={
                                  Object {
                                    "current": <button
                                      class="ui red fluid button"
                                      type="submit"
                                    >
                                      Log In
                                    </button>,
                                  }
                                }
                              >
                                <button
                                  className="ui red fluid button"
                                  onClick={[Function]}
                                  type="submit"
                                >
                                  Log In
                                </button>
                              </RefFindNode>
                            </Ref>
                          </Button>
                        </div>
                      </FormField>
                    </FormButton>
                  </form>
                </Form>
              </div>
            </CardContent>
          </div>
        </Card>
      </div>
    </LoginForm>
  </ConnectFunction>
</Provider>
`;
